backend/src/auth/services/auth_service.py

```diff
@@ -1,82 +1,225 @@
             raise InvalidPassword
+
+        # Return the authenticated user
         return user
 
-    def register(self, user: UserCreate, db: Session) -> Optional[RegisterResponse]:
+    def create_user(self, user: CreateUserModel, db: Session):
+        """
+        Creates a new user in the database.
+
+        Args:
+            user (CreateUserModel): The user model to be created.
+            db (Session): The database session.
+
+        Returns:
+            CreateUserModel: The created user model.
+
+        Raises:
+            HTTPException: If the user already exists in the database.
+        """
+        # Hash the user's password
         user.password = hash_password(user.password)
+
+        try:
+            # Create the user in the database
+            user = self.repository.create_user(user=user, db=db)
+        except Exception:
+            # If the user already exists, raise an HTTPException
+            raise HTTPException(status_code=400, detail="User already exists")
+
+        return user
+
+    @staticmethod
+    def publish_user_created_event(
+        user_id: str, user: CreateUserModel, profile: CreateProfileModel
+    ) -> None:
+        """
+        Publishes a user created event to the cache handler.
+
+        Args:
+            user_id (str): The ID of the user.
+            user (CreateUserModel): The user object containing user details.
+            profile (CreateProfileModel): The profile object containing profile details.
+        """
+        # Publish user created event to the cache handler
+        publish_handler_event(
+            pattern="user_created",
+            data={
+                "id": user_id,
+                "email": user.email,
+                "username": user.username,
+                "profile": {
+                    "name": profile.name,
+                    "surname": profile.surname,
+                    "image_url": profile.image_url,
+                },
+            },
+        )
+
+    def register(self, user: RegisterUserModel, db: Session) -> Optional[JSONResponse]:
+        """
+        Registers a new user and returns a JSONResponse.
+
+        Args:
+            user (RegisterUserModel): The user to register.
+            db (Session): The database session.
+
+        Returns:
+            Optional[JSONResponse]: The JSONResponse with the registration status.
+        """
+        # Check if the username is already taken
         if self.repository.get_by_username(user.username, db):
             raise UsernameTaken
+        # Check if the email is already taken
         elif self.repository.get_by_email(user.email, db):
             raise EmailTaken
+
         try:
-            self.repository.create_user(user, db)
-            return RegisterResponse(
-                message="User registered successfully",
+            # Create the user in the database
+            user_db = self.create_user(user=user, db=db)
```

Comment:
i would change it to `user_obj_db`, because `user_db` sounds like its database of user instead of object from database